paper/paper.md

```diff
@@ -0,0 +1,84 @@
+
+# Statement of Need
+
+The use of computational methods has become increasingly widespread in fields outside of computer science. As these disciplines require more computational tools, undergraduate curricula also begin to include topics in programming and computer science. However, because students are focused on their own discipline—and programming is likely a secondary interest—teaching students through traditional computer science offerings is not always effective [@Cortina2007; @Forte2005; @Guzdial2005]. While there are visual programming languages such as Raptor [@Carlisle2004] or Scratch [@Resnick2009] that are intended to be easy and enjoyable for non-computer science majors to learn, they lack the specialized tools that are required for effective work in domain sciences, such as numerical or data visualization libraries. A hybrid approach is to teach students computational concepts in an interactive environment where it is possible to quickly write, test, and tweak small pieces of code. Many such environments exist, including Mathematica [@mathematica], Maple [@maple], Matlab [@matlab], Sage [@sage] and IPython [@PerezGranger2007].
+
+In recent years the IPython project introduced the *Jupyter notebook* [@kluyver2016jupyter], an interface that is particularly conducive to interactive and literate computing where programmers can interleave prose with code and figures. The Jupyter notebook is ideal for educators because it allows them to create assignments which include instructions along with cells, in which students can provide solutions to exercises. Students can, for example, be asked to write code both to compute and visualize a particular result. Because of the interactive nature of the notebook, students can iterate on a coding problem without having to switch back and forth between the command line and a text editor, and they can see the results of their code almost instantly.
```

Comment:
- [ ] => `In recent years,`
- [ ] `particularly conducive` => `conducive`. Unqualified conduciveness is quieter, but stronger.
- [ ] `literature computing where` => `literate computing, where`
- [ ] This is the first time you are using the word "cells" I believe, so why don't you call them "code or Markdown cells" at least this once just to make sure everyone is on the same page.
- [ ] (Mildly pedantry). The ability to add code and Markdown cells for students does not provide the ability to *ask* the students to write code that computes and visualizes a results. It provides the ability for the students to do the code writing. (Remove `be asked to`.)
- [ ] "Because of the interactive nature of the notebook" => "Because the notebook is interactive".
- [ ] `having to` => `needing to`
- [ ] `the command line` => `a command line`
- [ ] `almost instantly` is a real bummer, like you're giving me just a little less than I really want — how about `quickly`?